vault/namespaces_store.go

```diff
@@ -738,7 +738,6 @@ func (ns *NamespaceStore) clearNamespaceResources(ctx context.Context, namespace
 
 	// Now grab write lock so that we can write to storage.
 	ns.lock.Lock()
```

Comment:
@wslabosz-reply I think this results in deadlock if we hit an error case in this function. 

```suggestion
	ns.lock.Lock()
	unlocked := false
	defer func() {
		if !unlocked {
			ns.lock.Unlock()
			unlocked = true
		}
	}()
```

and then below, you can call `ns.lock.Unlock()` and set `unlocked = true`.